### PR TITLE
[Merged by Bors] - feat: Supremum of `Finset.Iic a`

### PR DESCRIPTION
**Pull request by YaelDillies**
and lemmas about `Set.toFinset` of `Finset.Ixx`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/LocallyFinite.lean [hunk 1]
```diff
@@ -1470,3 +1470,65 @@ instance [Preorder α] [LocallyFiniteOrderTop α] : Finite { x : α // y < x } :
...
+end LocallyFiniteOrderBot
+end Set
+namespace Finset
+variable {α : Type*}
+
+section SemilatticeSup
+variable [SemilatticeSup α] [OrderBot α] [LocallyFiniteOrderBot α]
+
+@[simp] lemma sup_Iic (a : α) : (Iic a).sup id = a :=
+  le_antisymm (Finset.sup_le fun _ ↦ mem_Iic.1) $ le_sup (f := id) $ mem_Iic.2 $ le_refl a
```
Comment (eric-wieser) [inline Mathlib/Order/LocallyFinite.lean:1523]: Mathlib4 style guide says `<|` not `$`

File: Mathlib/Order/LocallyFinite.lean [hunk 2]
```diff
@@ -1470,3 +1470,65 @@ instance [Preorder α] [LocallyFiniteOrderTop α] : Finite { x : α // y < x } :
...
+
+@[simp]
+lemma toFinset_Iic (a : α) [Fintype (Iic a)] : (Iic a).toFinset = Finset.Iic a := by ext; simp
+
+@[simp]
+lemma toFinset_Iio (a : α) [Fintype (Iio a)] : (Iio a).toFinset = Finset.Iio a := by ext; simp
+
+end LocallyFiniteOrderBot
+end Set
+namespace Finset
```
Comment (eric-wieser) [inline Mathlib/Order/LocallyFinite.lean:1516]:
```suggestion
end Set

namespace Finset
```

File: Mathlib/Order/LocallyFinite.lean [hunk 3]
```diff
@@ -1470,3 +1470,65 @@ instance [Preorder α] [LocallyFiniteOrderTop α] : Finite { x : α // y < x } :
...
+
+@[simp] lemma sup_Iic (a : α) : (Iic a).sup id = a :=
+  le_antisymm (Finset.sup_le fun _ ↦ mem_Iic.1) $ le_sup (f := id) $ mem_Iic.2 $ le_refl a
+
+end SemilatticeSup
+
+section SemilatticeInf
+variable [SemilatticeInf α] [OrderTop α] [LocallyFiniteOrderTop α]
+
+@[simp] lemma inf_Ici (a : α) : (Ici a).inf id = a :=
```
Comment (eric-wieser) [inline Mathlib/Order/LocallyFinite.lean:1530]: Do you think the `inf'` version is worth having too?

File: Mathlib/Order/LocallyFinite.lean [hunk 4]
```diff
@@ -1470,3 +1470,65 @@ instance [Preorder α] [LocallyFiniteOrderTop α] : Finite { x : α // y < x } :
...
   rfl
+
+namespace Set
+variable {α : Type*} [Preorder α]
+
+section LocallyFiniteOrder
+variable [LocallyFiniteOrder α]
+
+@[simp] lemma toFinset_Icc (a b : α) [Fintype (Icc a b)] : (Icc a b).toFinset = Finset.Icc a b := by
+  ext; simp
```
Comment (eric-wieser) [inline Mathlib/Order/LocallyFinite.lean:1481]: Please mention these lemmas in the PR description.

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Data/Finset/LocallyFinite.lean [hunk 5]
```diff
@@ -67,6 +67,7 @@ theorem nonempty_Ioc : (Ioc a b).Nonempty ↔ a < b := by
   rw [← coe_nonempty, coe_Ioc, Set.nonempty_Ioc]
 #align finset.nonempty_Ioc Finset.nonempty_Ioc
 
+-- TODO: This is nonsense. A locally finite order is never densely ordered
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/LocallyFinite.lean:70]: Can you ping the author of this lemma (and link the PR that added it) and check that they agree before merging?

**Review by eric-wieser — COMMENTED** [review]
bors d+

(since it failed last time)

File: Mathlib/Data/Finset/LocallyFinite.lean (hunk 5 above)
Comment (YaelDillies) [reply Mathlib/Data/Finset/LocallyFinite.lean:70]: I was the author in https://github.com/leanprover-community/mathlib/pull/9495 😅

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
Turns out the `Set.Finite` lemmas can already be proved by `simp`.

bors merge

**Comment by YaelDillies** [comment]
bors merge

File: Mathlib/Data/Finset/LocallyFinite.lean (hunk 5 above)
Comment (b-mehta) [reply Mathlib/Data/Finset/LocallyFinite.lean:70]: Do we have the statement that this is nonsense? Perhaps in the form that intervals in a dense order are infinite

**Comment by YaelDillies** [comment]
Ugh those simpNF errors are weird.

bors merge

**Pull request closed without merging**